estuary-cdk/estuary_cdk/capture/common.py

```diff
@@ -552,6 +568,10 @@ async def _binding_backfill_task(
             if isinstance(item, BaseDocument):
                 task.captured(binding_index, item)
                 done = True
+            elif isinstance(item, AssociatedDocument):
+                # TODO(jshearer): Assert that the binding index exists (and is enabled?)
```

Comment:
nit: disabled bindings are not represented in the built CaptureSpec and cannot be present in `resolved_bindings`

Reply:
What does the runtime do when it encounters a document for a binding index that doesn't exist? You could theoretically just `yield AssociatedDocument(doc=..., binding=999999)`. I suppose if the runtime errors if it sees an invalid binding index then extra validation here is unneccesary

Reply:
https://github.com/estuary/flow/blob/master/crates/runtime/src/capture/protocol.rs#L371-L375